exe/wallet/jormungandr/Main.hs

```diff
@@ -77,197 +81,270 @@ import Cardano.Wallet.Unsafe
     ( unsafeRunExceptT )
 import Cardano.Wallet.Version
     ( showVersion, version )
+import Control.Applicative
+    ( optional )
 import Control.Concurrent.Async
     ( race_ )
+import Control.Monad
+    ( join )
 import Data.Coerce
     ( coerce )
 import Data.Function
     ( (&) )
-import Data.Proxy
-    ( Proxy (..) )
 import Data.Text
     ( Text )
 import Data.Text.Class
-    ( ToText (..) )
+    ( ToText (..), showT )
 import Network.HTTP.Client
-    ( Manager, defaultManagerSettings, newManager )
+    ( defaultManagerSettings, newManager )
 import Network.Wai.Handler.Warp
     ( setBeforeMainLoop )
+import Options.Applicative
+    ( CommandFields
+    , Mod
+    , Parser
+    , command
+    , execParser
+    , help
+    , helper
+    , info
+    , long
+    , metavar
+    , progDesc
+    )
 import Servant.Client
     ( BaseUrl (..), Scheme (..) )
-import System.Console.Docopt
-    ( Docopt, longOption )
-import Text.Heredoc
-    ( here )
+import System.FilePath
+    ( (</>) )
 
 import qualified Cardano.Wallet as Wallet
 import qualified Cardano.Wallet.Api.Server as Server
 import qualified Cardano.Wallet.DB.Sqlite as Sqlite
 import qualified Cardano.Wallet.Jormungandr.Network as Jormungandr
 import qualified Cardano.Wallet.Jormungandr.Transaction as Jormungandr
+import qualified Data.ByteString.Lazy as BL
 import qualified Data.Text as T
 import qualified Network.Wai.Handler.Warp as Warp
 
 {-------------------------------------------------------------------------------
                               Main entry point
+
+  FIXME:
+  We instiantiate everything below to 'Testnet' because there's no way
+  to actually implement the wallet, transaction and address command without
+  any prior knowledge of the target network (since the serialization of
+  addresses depends on that).
+
+  This is okay for now as do only have access to a 'Testnet' anyway, but in the
+  long run, we may want to (pick one):
+
+  - Provide a `--network` parameter to pretty much every command
+  - Construct two executables, one for 'Testnet' and one for 'Mainnet'
+  - Use an environment variable
+  - ?
```

Comment:
Currently, when the CLI is used in client mode (i.e., called with `wallet`, `transaction`, or `address`), it doesn't know whether the server is running with `Testnet` or `Mainnet`.

However, on the other hand, the server _does_ know this information.

**Couldn't the CLI first call the wallet API server to find out?** 

Of course, we'd have to provide an endpoint to allow this query. It would also mean that these commands require two API calls rather than one, but that shouldn't really affect the user experience.

---

As for the other solutions:

1. **Providing a `--network` parameter to pretty much every command**. I think this is a bit awkward from a usability perspective. From the user's perspective, they have "already told" the wallet server which network they want. It seems strange to require it again. It also creates the possibility for errors: what would happen if the user launched with `--network testnet`, but then ran commands with `--network mainnet`? Would they see serialization errors?

2. **Construct two executables.** This option seems like it could add unnecessary complexity to our build system, documentation, and testing.

3. **Use an environment variable.** I think this feels like a "mutable global variable". :) What if the user started the server in one shell (where `NETWORK=mainnet`) and then ran commands in another shell (where NETWORK=testnet`)? (This is a similar situation to solution #1 above.)

Reply:
Well, 1. and 3. are more or less the same indeed although, the ENV var is more user-friendly as it can be exported once in a terminal and then, used implicitly. 
About the discrepancy between how the server is started and what network is used for each command, that's something we will be facing _anyway_. If someone started a server as `mainnet` and submit `testnet` addresses, one will end up with an error from the API about the address format.

Another option could be to actually figure out the network ourselves from the arguments. The network type enters in play only for the address discrimination (at the API-level I mean). And, it's not possible to mix them. So, when parsing the arguments, we can also check to what network do addresses belong. If there's a mix, we fail. If not, we submit them as-is and, the server will react accordingly.

Reply:
> Another option could be to actually figure out the network ourselves from the arguments. The network type enters in play only for the address discrimination (at the API-level I mean). And, it's not possible to mix them. So, when parsing the arguments, we can also check to what network do addresses belong. If there's a mix, we fail. If not, we submit them as-is and, the server will react accordingly.

This sounds like a good option. Hopefully we can produce a sensible error message if the server rejects a given address due to it belonging to the wrong network.

Reply:
> Hopefully we can produce a sensible error message if the server rejects a given address due to it belonging to the wrong network.

I think the server already produces good-enough error message regarding addresses :sweat_smile: 

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs#L167-L174

At least, that's for Jörmungandr. I am actually not sure that the implementation from the bridge would reject addresses from a wrong network :thinking: ...